File: Codewars/Python/Kyu5_Valid Parentheses.py
def valid_parentheses(string):
    open_bracket,closed_bracket = 0,0
    if string == '':
        return True
    elif string[0] == ")" or string[-1] == "(":return False
    for i in string:
        if closed_bracket > open_bracket:return False
        if i == "(":
            open_bracket += 1
        elif i ==")":
            closed_bracket += 1
    if open_bracket == closed_bracket:return True
    else:return False

File: Codewars/Go/Kyu5_Valid Parentheses.go
package kata

func ValidParentheses(parens string) bool {
  var openBracket,closedBracket int = 0,0
  
  if parens == "" {
    return true
  } else if string(parens[0]) == ")" || string(parens[len(parens)-1]) == "(" {
    return false
  }
  
  for _, char := range parens {
    if closedBracket > openBracket {
      return false
    }
    
    switch char {
      case '(':
        openBracket++
      case ')':
        closedBracket++
    
    }
  }
  
  return openBracket == closedBracket
}